packages/analytics/src/functions.ts

```diff
@@ -137,6 +138,34 @@ export async function setUserProperties(
   }
 }
 
+/**
+ * Retrieves a unique Google Analytics identifier for the web client.
+ * See {@link https://developers.google.com/analytics/devguides/collection/ga4/reference/config#client_id | client_id}.
+ *
+ * @public
+ *
+ * @param gtagFunction Wrapped gtag function that waits for fid to be set before sending an event
+ */
+export async function internalGetGoogleAnalyticsClientId(
+  gtagFunction: Gtag,
+  initializationPromise: Promise<string>
+): Promise<string> {
+  const measurementId = await initializationPromise;
+  return new Promise((resolve, reject) => {
+    gtagFunction(
+      GtagCommand.GET,
+      measurementId,
+      'client_id',
+      (fieldName: string) => {
```

Comment:
I didn't realize this at first, but this is the client id right? Why is named "fieldName"?

Reply:
I named it field name because of the docs https://developers.google.com/tag-platform/gtagjs/reference#get but because we're just using clientId it makes sense to update it